### PR TITLE
[DROOLS-7125] Drools 8 docs : throughout review. Fix broken links, outdated description

### PR DESCRIPTION
**Pull request by tkobayas**
- Remove mention about Spring and OSGi
- Update pom.xml screenshot
- Remove KIE Server related sentences
- Fix broken links

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-7125

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-lts</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>

 - for <b>mandrel lts checks</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins run mandrel-lts</b>

- for a <b>specific mandrel lts check</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel-lts</b>
</details>

### CODE REVIEW

File: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_KIEModuleIntroduction-section.adoc [hunk 1]
```diff
@@ -16,7 +15,7 @@ The plugin also generates many classes, making the runtime loading faster too.
 The example project layout and Maven POM descriptor is illustrated in the screenshot
 
 .Example project layout and Maven POM
-image::KIE/BuildDeployUtilizeAndRun/defaultkiesession.png[align="center"]
+image::KIE/BuildDeployUtilizeAndRun/project-layout.png[align="center"]
```
Comment (tkobayas) [inline drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_KIEModuleIntroduction-section.adoc:18]: Updated the screenshot.

File: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_kie-scanner-proc.adoc [hunk 2]
```diff
@@ -50,12 +50,12 @@ The project group ID, artifact ID, and version (GAV) settings in the example are
...
-  <id>guvnor-m2-repo</id>
+  <id>my-nexus-env</id>
   <repositories>
     <repository>
-      <id>guvnor-m2-repo</id>
-      <name>BA Repository</name>
-      <url>http://localhost:8080/\{URL_COMPONENT_CENTRAL}/maven2/</url>
+      <id>my-nexus</id>
+      <name>My Nexus repository</name>
+      <url>http://repository.example.org/nexus/content/groups/public/</url>
```
Comment (tkobayas) [inline drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_kie-scanner-proc.adoc:58]: This part was originally assuming an architecture where KIE Server refers to a repository in Business Central. Changed it to an architecture where a plain KieScanner refers to a user repository (e.g. nexus).

File: drools-docs/src/modules/ROOT/pages/rule-engine/_performance-tuning-rule-engine-ref.adoc [hunk 3]
```diff
@@ -34,7 +34,7 @@ try {
 }
 ----
 
-For information about built-in event listeners and debug logging in the {RULE_ENGINE}, see xref:engine-event-listeners-con_rule-engine[].
+For information about built-in event listeners and debug logging in the {RULE_ENGINE}, see xref:#engine-event-listeners-con_rule-engine[{RULE_ENGINE_CAP} event listeners and debug logging].
```
Comment (tarilabs) [inline drools-docs/src/modules/ROOT/pages/rule-engine/_performance-tuning-rule-engine-ref.adoc]: just a comment, `RULE_ENGINE_CAP` looks the same value as `RULE_ENGINE`, do you want to fold them into 1 ?

https://github.com/kiegroup/drools/blob/5306c5fa3be41f0ad45734d15e43dc26435f7298/drools-docs/src/modules/ROOT/pages/_artifacts/document-attributes-drools.adoc#L21-L22

I believe originally one was used for shortref and another for "being in the docs'.
But imho we can just keep `:RULE_ENGINE: Drools rule engine` and use that one, wdyt ?

**Review by tarilabs — APPROVED** [review]
minor optional comment which could also be done separately, thank you for this review Toshiya!

File: drools-docs/src/modules/ROOT/pages/rule-engine/_performance-tuning-rule-engine-ref.adoc (hunk 3 above)
Comment (tkobayas) [reply drools-docs/src/modules/ROOT/pages/rule-engine/_performance-tuning-rule-engine-ref.adoc]: Thank you for the suggestion! Agreed. Updated this PR.

**Review by kaldesai — APPROVED** [review]
Your changes look good to me! Thank you!

**Comment by tkobayas** [comment]
@tarilabs Please merge this PR, thanks!

**Pull request merged**